packages/wrangler/src/config/index.ts

```diff
@@ -55,6 +57,12 @@ export function readConfig<CommandArgs>(
 	 * so we should error if `wrangler.json` is detected in a Pages project
 	 */
 	const isPagesConfig = rawConfig.pages_build_output_dir !== undefined;
```

Comment:
By the way, there is actually a helper function for this in `validation.ts` - `isPagesConfig(rawConfig)` - which we should use.